validator/validation/batches.py

```diff
@@ -118,21 +157,108 @@ def reshape_meta(meta):
         for network, net_ids in ids.items():
             lons, lats, meta_list = [], [], []
             for idx in net_ids:
-                meta = ref_reader.read_ts(idx, return_meta=True)[1]
+                meta = reader.read_ts(idx, return_meta=True)[1]
                 meta = reshape_meta(meta)
                 lons.append(meta['longitude'])
                 lats.append(meta['latitude'])
                 meta_list.append(meta)
             gpis = net_ids
             gpis, lons, lats = np.array(gpis), np.array(lons), np.array(lats)
 
-            gpis, lons, lats, index = _geographic_subsetting(gpis, lons, lats, validation_run.min_lat, validation_run.min_lon, validation_run.max_lat, validation_run.max_lon)
+            gpis, lons, lats, index = _geographic_subsetting(
+                gpis, lons, lats, validation_run.min_lat,
+                validation_run.min_lon, validation_run.max_lat, validation_run.max_lon
+            )
 
             if len(gpis) > 0:
                 jobs.append((gpis, lons, lats, meta_list))
                 total_points += len(gpis)
 
     else:
-        raise ValueError("Don't know how to get gridpoints and generate jobs for reader {}".format(ref_reader))
+        raise ValueError("Don't know how to get gridpoints and generate jobs for reader {}".format(reader))
+
+    if not return_points:
+        return jobs
 
     return total_points, jobs
+
+
+def create_upscaling_lut(
+        validation_run,
+        datasets,
+        ref_name
+) -> dict:
+    """
+    Create a lookup table that aggregates the non-reference measurement points falling under the same reference
+    pixel
+
+    Parameters
+    ----------
+    validation_run: models.validation_run.ValidationRun object
+        Configuration of the run
+    datasets : dict of dicts
+        :Keys: string, datasets names
+        :Values: dict, containing the following fields (see pytesmo DataManager for details):
+            *'class'
+            *'columns'
+            *'args': list, optional
+            *'kwargs': dict, optional
+            *'grids_compatible': boolean, optional
+            *'use_lut': boolean, optional
+            *'lut_max_dist': float, optional
+    ref_name: str
+        Name of the reference dataset
+
+    Returns
+    -------
+    lut: dict
+        lookup table with shape {'other_dataset':{ref gpi: [other gpis]}}
+    """
+    ref_reader = datasets[ref_name]["class"]
+    # get grid of reference
+    while hasattr(ref_reader, 'cls'):
+        ref_reader = ref_reader.cls
```

Comment:
This should not be necessary anymore

Reply:
You mean after your PR?

Reply:
The tests fail when I don't use this

Reply:
It should be fixed on pytesmo 0.11.2 or 0.11.3. What's the error in the tests?

Reply:
`AttributeError: 'TimezoneAdapter' object has no attribute 'grid'`